Modules/_ctypes/_ctypes.c

```diff
@@ -1351,7 +1351,7 @@ PyCArrayType_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     length = PyLong_AsLongAndOverflow(length_attr, &overflow);
     if (overflow) {
         PyErr_SetString(PyExc_OverflowError,
-                        "The '_length_' attribute is too large");
+                        "The '_length_' attribute does not fit in C long");
```

Comment:
on second thought, this changes the behavior from raising OverflowError for small negative `_length_` to raising ValueError.
should I open another sub-issue of bpo-29833 for that?
or maybe should I just raise OverflowError with a better error message?